src/main/java/bitflow4j/steps/database/WriteToInfluxDB.java

```diff
@@ -54,22 +56,64 @@ public WriteToInfluxDB(String databaseURL, String databaseName, String username,
 
     @Override
     public void writeSample(Sample sample) throws IOException {
-        checkConnectionAndReconnect();
-
-        // Build database point
-        String finalPrefix = sample.resolveTagTemplate(prefix);
-        Point.Builder pointBuilder = Point.measurement(finalPrefix)
-                .time(sample.getTimestamp().getTime(), TimeUnit.MILLISECONDS);
-        for (int j = 0; j < sample.getMetrics().length; j++) {
-            pointBuilder.addField(sample.getHeader().header[j], sample.getValue(j));
+        try {
+            checkConnectionAndReconnect();
+            ensureDefaultDatabase();
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, String.format("Failed to connect to InfluxDB at %s: %s", databaseURL, buildExceptionMessage(e)));
+            return;
+        }
+
+        // Build and send database points.
+        // Split the sample into parts to avoid parser errors due to  overly long request.
+        for (int i = 0; i < sample.getMetrics().length; i += metricsPerRequest) {
+            String finalPrefix = sample.resolveTagTemplate(prefix);
+
+            Point.Builder pointBuilder = Point.measurement(finalPrefix)
+                    .time(sample.getTimestamp().getTime(), TimeUnit.MILLISECONDS);
+            for (int j = i; j < sample.getMetrics().length && j < i + metricsPerRequest; j++) {
+                double val = filterInvalidValues(sample.getValue(j));
+                pointBuilder.addField(sample.getHeader().header[j], val);
+            }
+
+            // Send database point
+            Point point = pointBuilder.build();
+            if (logger.isLoggable(Level.FINER)) {
+                logger.finer(String.format("Sending point to InfluxDB at %s, line protocol length: %s", databaseURL, point.lineProtocol().length()));
+            }
+            influxDB.write(point);
         }
-        Point point = pointBuilder.build();
-        // Send database point
-        influxDB.write(point);
 
         output.writeSample(sample);
     }
 
+    private static String buildExceptionMessage(Throwable e) {
+        StringBuilder b = new StringBuilder();
+        while (e != null) {
+            b.append("[");
+            b.append(e.getClass().getName());
+            b.append("] ");
+            b.append(e.getMessage());
+            e = e.getCause();
+            if (e != null)
+                b.append(", caused by: ");
+        }
+        return b.toString();
+    }
+
+    private static double filterInvalidValues(double val) {
```

Comment:
This is dangerous. Wouldn't it be better to make this function return a boolean and let influxDB skip this metric entry? InfluxDB does not need to receive every metric for every datapoint. In Grafana this would mean you have some "holes" in your time series but this is much better than inserting values into the DB which were not present like this, especially Double.MAX_VALUE for Double.NaN.